### PR TITLE
v1.14 Backports 2024-04-02

### PR DESCRIPTION
**Pull request by joamaki**
* [x] #29300 (@learnitall)
 * [x] #31488 (@bimmlerd)
 * [x] #31478 (@julianwiedmann)
 * [x] #31594 (@qmonnet)
 * [x] #31635 (@pchaigno)
 * [x] #31622 (@gandro)
 * [x] #31627 (@pchaigno)
 * [x] #31104 (@tamilmani1989)
 * [x] #31641 (@tklauser)
 * [x] #31652 (@qmonnet)

PRs skipped due to conflicts:

 * #31407 (@tommyp1ckles)
 * #31328 (@nathanjsweet)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 29300 31488 31478 31594 31635 31622 31627 31104 31641 31652
```

### CODE REVIEW

**Comment by joamaki** [comment]
@bimmlerd @tklauser same type of conflicts as with v1.13 here.

**Comment by joamaki** [comment]
@tommyp1ckles skipped the node-map-max as the HeaderWriter wasn't modularized. Could you take a look at doing a manual backport for it?

@nathanjsweet skipped as proxyPolicy struct didn't exist. Could you take a look at whether we need to backport more commits or if there's another way to do it?

**Review by bimmlerd — APPROVED** [review]

**Comment by joamaki** [comment]
/test-backport-1.14

**Review by qmonnet — APPROVED** [review]
Thanks!

**Review by tklauser — APPROVED** [review]
Looks good for my changes, thanks.

**Review by learnitall — APPROVED** [review]
Thanks!

**Review by pchaigno — CHANGES_REQUESTED** [review]
There's the same issue as for v1.13: https://github.com/cilium/cilium/pull/31627 wasn't backported because the file moved. The code to edit can be found in `.github/workflows/conformance-ipsec-e2e.yaml`.

File: .github/workflows/conformance-ipsec-e2e.yaml [hunk 1]
```diff
@@ -322,7 +322,7 @@ jobs:
               if [[ $keys_in_use == $exp_nb_keys ]]; then
                 break
               fi
-              echo "Waiting until key rotation starts (seeing $keys_in_use keys)"
+              echo "Waiting until key rotation completes (seeing $keys_in_use keys, expected $exp_nb_keys)"
```
Comment (pchaigno) [inline .github/workflows/conformance-ipsec-e2e.yaml]: Same here regarding `starts` -> `completes`.

Comment (joamaki) [reply .github/workflows/conformance-ipsec-e2e.yaml]: Fixed here as well.

**Review by pchaigno — APPROVED** [review]
All good for my PRs. Thanks!

**Review by julianwiedmann — APPROVED** [review]
@tklauser maybe you could have a look at Sebastian's PR as well? :pray:

**Review by tklauser — APPROVED** [review]
> @tklauser maybe you could have a look at Sebastian's PR as well? 🙏

Sure. Backport of #31622 looks good to me, thanks!

**Comment by joamaki** [comment]
/test-backport-1.14

**Pull request merged**